src/Widgets.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Bolt;
+
+use Bolt\Snippet\QueueProcessor;
+use Bolt\Snippet\RequestZone;
+use Bolt\Widget\RequestAware;
+use Bolt\Widget\TwigAware;
+use Bolt\Widget\WidgetInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpFoundation\Response;
+use Tightenco\Collect\Support\Collection;
+use Twig\Environment;
+
+/**
+ * @todo Determine if we should split this up into smaller classes
+ */
+class Widgets
+{
+    /** @var Collection */
+    private $queue;
+
+    /** @var RequestStack */
+    private $requestStack;
+
+    /** @var QueueProcessor */
+    private $queueProcessor;
+
+    /** @var Environment */
+    private $twig;
+
+    public function __construct(RequestStack $requestStack, QueueProcessor $queueProcessor, Environment $twig)
+    {
+        $this->queue = new Collection([]);
+        $this->requestStack = $requestStack;
+        $this->queueProcessor = $queueProcessor;
+        $this->twig = $twig;
+    }
+
+    public function registerWidget(WidgetInterface $widget): void
+    {
+        if ($widget instanceof RequestAware) {
+            $widget->setRequest($this->requestStack->getCurrentRequest());
```

Comment:
request should not be set when registering, but right before invoking/rendering (or pass whole RequestStack)

Reply:
Changed!